onnxruntime/core/optimizer/qdq_transformer/qdq_util_minimal.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <functional>
+#include <string>
+
+namespace ONNX_NAMESPACE {
+class TensorProto;
+}
+
+namespace onnxruntime {
+
+class Node;
+class Path;
+
+namespace QDQ {
+
+constexpr const char* QOpName = "QuantizeLinear";
+constexpr const char* DQOpName = "DequantizeLinear";
+
+enum InputIndex : int {
+  INPUT_ID = 0,
+  SCALE_ID = 1,
+  ZERO_POINT_ID = 2,
+  TOTAL_COUNT = 3,
+};
+
+// Check if Q/DQ pair is supported in the QDQ transformer. It requires:
+// 1. Q/DQ doesn't have optional input.
+// 2. scale and zero point is constant scalar
+// 3. Q and DQ have same scale and zero point
+bool IsQDQPairSupported(
+    const Node& q_node, const Node& dq_node,
+    const std::function<const ONNX_NAMESPACE::TensorProto*(const std::string&)>& get_const_initializer,
+    const Path& model_path);
+
+// Check if DQ is supported in the QDQ transformer. It requires:
+// 1. DQ doesn't have optional input.
+// 2. scale and zero point is constant scalar
```

Comment:
not understanding this limitation. the zero point value is the only optional input, and if missing defaults to 0. why do we care if it's there or not?

Reply:
code in selector assumes no optional input, so we add the constraint here. And we will insert a zero point manually if it is optional.

Reply:
when all the QDQ transformers are at level 2 this makes sense. but now with the QDQ propagation one moving to level 1 I'll update it to handle missing zero point input.